src/customers.query.ts

```diff
@@ -1,28 +1,25 @@
 import { Knex } from "knex"
-import { Customer } from "./dbmodel"
+import { Customer, CustomerKpi } from "./models"
 
-export async function get_customers_list(db: Knex): Promise<Customer[]>
-{
-    const customers_list: Customer[] = await db.select("username", "lastname", "birthdate",
-        "kpis.number_purchase", "kpis.store", "kpis.status").from("customers").leftJoin("kpis",
-        "kpis.customer_id", "customers.id")
+export async function get_customers_list(db: Knex): Promise<CustomerKpi[]> {
+  const customers_list: CustomerKpi[] = await db("kpis")
+    .select(
+      "customers.id",
+      "customers.username",
+      "customers.lastname",
+      "customers.birthdate",
+      "number_purchase",
+      "store",
+      "status"
+    )
+    .fullOuterJoin("customers", "customers.id", "kpis.customer_id")
```

Comment:
**Should have:** On ne devrait pas avoir besoin d'un full outer join, de ce que je comprends ça prend tout de la table à gauche et tout de la table à droite.
https://www.w3schools.com/sql/sql_join_full.asp

Il faudrait utiliser un `.leftJoin`, prendre les clients, et les kpis de ces clients ou rien.

Reply:
Du tout. J'ai essayé avec le "leftJoin," mais rien n'a été affiché.